content/guides/azure-pipelines.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Introduction to Azure Pipelines with Docker
+linkTitle: Azure Pipelines and Docker
+summary: |
+  Learn how to automate Docker image build and push using Azure Pipelines.
+params:
+  tags: [devops]
+  time: 10 minutes
+---
+
+## Prerequisites
+
+Before you begin, ensure the following requirements are met:
+
+- A [Docker Hub account](https://hub.docker.com) with a generated access token.
+- An active [Azure DevOps project](https://dev.azure.com/) with a connected [Git repository](https://learn.microsoft.com/en-us/azure/devops/repos/git/?view=azure-devops).
+- A project that includes a valid [`Dockerfile`](https://docs.docker.com/engine/reference/builder/) at its root or appropriate build context.
+
+## Overview
+
+This guide walks you through building and pushing Docker images using [Azure Pipelines](https://azure.microsoft.com/en-us/products/devops/pipelines), enabling a streamlined and secure CI workflow for containerized applications. You’ll learn how to:
+
+- Configure Docker authentication securely.
+- Set up an automated pipeline to build and push images.
+
+## Step 1: Configure Docker credentials
+
+To authenticate securely with Docker Hub:
+
+1. In your Azure DevOps project, navigate to **Project Settings > Pipelines > Library**.
+2. Create a new **Variable Group** and add the following variables:
```

Comment:
More secure method would be using service connection. Good to mentioned it as step need to follow before creating the CI 

https://learn.microsoft.com/en-us/azure/devops/pipelines/library/service-endpoints?view=azure-devops#docker-hub-or-others